sunpy/net/scraper.py

```diff
@@ -386,15 +324,16 @@ def _check_timerange(self, url, timerange):
         `bool`
             `True` if URL's valid time range overlaps the given timerange, else `False`.
         """
-        if hasattr(self, 'extractor'):
-            exdict = parse(self.extractor, url).named
-            tr = get_timerange_from_exdict(exdict)
-            return tr.intersects(timerange)
-        else:
-            datehref = self._extractDateURL(url).to_datetime()
-            smaller_pattern = self._smallerPattern(self.pattern)
-            file_timerange = TimeRange(datehref, datehref + smaller_pattern)
-            return file_timerange.intersects(timerange)
+        exdict = parse(self.pattern, url).named
+        if exdict['year'] < 100:
```

Comment:
Does a client hit this block of code?

Reply:
yes, the [STEREO HET](https://github.com/sunpy/sunpy/blob/813592baab6181288d8d50d6676d585374554960/sunpy/net/tests/test_scraper.py#L198), the files look like [this](https://izw1.caltech.edu/STEREO/DATA/HET/Ahead/1minute/)